llvm/lib/Target/AArch64/AArch64TargetTransformInfo.cpp

```diff
@@ -1174,7 +1174,8 @@ static std::optional<Instruction *> instCombineSVECmpNE(InstCombiner &IC,
     return std::nullopt;
 
   // Where the dupq is a lane 0 replicate of a vector insert
-  if (!cast<ConstantInt>(DupQLane->getArgOperand(1))->isZero())
+  auto *DupQLaneOp1 = dyn_cast<ConstantInt>(DupQLane->getArgOperand(1));
```

Comment:
Up to you but I think `DupQLaneIdx` would be more readable.

Reply:
Ok, fixed.